controller/konnect/konnectextension_controller_utils.go

```diff
@@ -0,0 +1,81 @@
+package konnect
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+func getKonnectAPIAuthRefNN(ctx context.Context, cl client.Client, ext *konnectv1alpha1.KonnectExtension) (types.NamespacedName, error) {
+	if ext.Spec.KonnectConfiguration != nil {
+		// TODO: handle cross namespace refs when supported.
+		return types.NamespacedName{
+			Namespace: ext.Namespace,
+			Name:      ext.Spec.KonnectConfiguration.APIAuthConfigurationRef.Name,
+		}, nil
+	}
+
+	// In case the KonnectConfiguration is not set, we fetch the KonnectGatewayControlPlane
+	// and get the KonnectConfiguration from `spec.konnectControlPlane.controlPlane.konnectNamespacedRef`.
+	// KonnectGatewayControlPlane reference and KonnectConfiguration
+	// are mutually exclusive in the KonnectExtension API.
+	cpRef := ext.Spec.KonnectControlPlane.ControlPlaneRef.KonnectNamespacedRef
+	kgcp := &konnectv1alpha1.KonnectGatewayControlPlane{}
+	err := cl.Get(ctx, client.ObjectKey{
+		// TODO: handle cross namespace refs to KonnectGatewayControlPlane when referencing CP from another namespace is supported.
+		Namespace: ext.Namespace,
+		Name:      cpRef.Name,
+	}, kgcp)
+	if err != nil {
+		return types.NamespacedName{}, err
+	}
+	return types.NamespacedName{
+		Namespace: kgcp.Namespace,
+		Name:      kgcp.Spec.KonnectConfiguration.APIAuthConfigurationRef.Name,
+		// NOTE: We requeue here to keep enforcing the state of the resource in Konnect.
+		// Konnect does not allow subscribing to changes so we need to keep pushing the
+		// desired state periodically.
```

Comment:
?

Reply:
Copy-pasta from the generic reconciler. Deleting it.